### PR TITLE
ACM 2113 -- Overview page cluster add on statuses

### PR DESCRIPTION
**Pull request by jeswanke**
Signed-off-by: John Swanke <jswanke@redhat.com>

### CODE REVIEW

**Review by KevinFCormier — CHANGES_REQUESTED** [review]
A few nits in inline comments.

I am also wondering if while you are in this code, you can make the tiles consistent. Currently, the **Cluster issues** tile does not provide a search link if there are 0 items, but all other tiles do. I think it's probably better to always provide the link. What do you think?

File: frontend/src/routes/Infrastructure/Clusters/ClusterSets/components/useClusterAddons.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/* Copyright Contributors to the Open Cluster Management project */
```
[the remark this answers is not in the document]
Comment (jeswanke) [reply frontend/src/routes/Infrastructure/Clusters/ClusterSets/components/useClusterAddons.tsx:1]: I put it next to the useClusters and useCanJoinClusterSets hooks :)

File: frontend/src/routes/Home/Overview/OverviewPage.tsx [hunk 2]
```diff
@@ -160,6 +197,13 @@ export default function OverviewPage() {
     regions: new Set(),
     ready: 0,
     offline: 0,
+    addons: {
+      healthy: { cnt: 0, clusters: new Set() },
+      danger: { cnt: 0, clusters: new Set() },
+      progress: { cnt: 0, clusters: new Set() },
+      pending: { cnt: 0, clusters: new Set() },
+      Unknown: { cnt: 0, clusters: new Set() },
```
[the remark this answers is not in the document]
Comment (jeswanke) [reply frontend/src/routes/Home/Overview/OverviewPage.tsx:205]: changed the cnt and Unknown

File: frontend/src/ui-components/AcmCharts/AcmDonutChart/AcmDonutChart.tsx [hunk 3]
```diff
@@ -143,10 +143,18 @@ const importantColorClass = 'var(--pf-chart-color-orange-300)'
...
 const lowColorClass = 'var(--pf-chart-color-blue-200)'
 const successColorClass = 'var(--pf-chart-color-black-100)'
+const unknownColorClass = 'var(--pf-chart-color-black-300)'
 
 export const colorThemes = {
   criticalImportantSuccess: [criticalColorClass, importantColorClass, successColorClass],
   criticalSuccess: [criticalColorClass, successColorClass],
   criticalLowSuccess: [criticalColorClass, moderateColorClass, successColorClass],
   criticalImportantModerateLow: [criticalColorClass, importantColorClass, moderateColorClass, lowColorClass],
+  criticalImportantPendingSuccess: [
```
[the remark this answers is not in the document]
Comment (jeswanke) [reply frontend/src/ui-components/AcmCharts/AcmDonutChart/AcmDonutChart.tsx]: fixed

File: frontend/src/routes/Home/Overview/OverviewPage.tsx [hunk 4]
```diff
@@ -381,23 +429,23 @@ export default function OverviewPage() {
 
   // TODO: Breaks url if length of selectedClustersFilter is too big.
   // Issue: https://github.com/open-cluster-management/backlog/issues/7087
-  function buildClusterComplianceLinks(clusterNames: Array<string> = []): string {
+  function buildClusterLinks(clusterNames: Array<string> = [], related: string): string {
```
[the remark this answers is not in the document]
Comment (jeswanke) [reply frontend/src/routes/Home/Overview/OverviewPage.tsx]: fixed

File: frontend/src/routes/Home/Overview/OverviewPage.tsx [hunk 5]
```diff
@@ -418,40 +466,64 @@ export default function OverviewPage() {
...
+        isPrimary: true,
+        link: buildClusterLinks(Array.from(healthy.clusters), 'ClusterManagementAddOn'),
+      },
+    ]
+  }, [healthy, danger, progress, pending, unknown, t])
+
   const policyReportData = useMemo(() => {
+    const getLink = (type: any, count: number) => {
+      return `${NavigationPath.search}?filters={"textsearch":"kind%3APolicyReport%20${type}%3A${
+        count > 0 ? '>' : '='
```
Comment (KevinFCormier) [inline frontend/src/routes/Home/Overview/OverviewPage.tsx:503]: I don't think we should change the query depending on the current count. The tile says "there are 0 clusters with critical issues" not "these are the clusters with 0 critical issues".

Comment (jeswanke) [reply frontend/src/routes/Home/Overview/OverviewPage.tsx:503]: cool--what should the query be?

Comment (KevinFCormier) [reply frontend/src/routes/Home/Overview/OverviewPage.tsx:503]: Just the version with `>0` always.

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/ClusterDetails/ClusterDetails.tsx [hunk 6]
```diff
@@ -127,8 +121,7 @@ export default function ClusterDetailsPage({
   const managedClusterInfo = managedClusterInfos.find(
     (mci) => mci.metadata?.name === name && mci.metadata?.namespace === name
   )
-  const clusterAddons = managedClusterAddons.filter((mca) => mca.metadata?.namespace === name)
```
Comment (KevinFCormier) [inline frontend/src/routes/Infrastructure/Clusters/ManagedClusters/ClusterDetails/ClusterDetails.tsx:130]: image

**Pull request merged**